core/src/main/java/org/elasticsearch/http/netty/NettyHttpChannel.java

```diff
@@ -134,6 +137,7 @@ public void sendResponse(RestResponse response) {
             if (!addedReleaseListener && content instanceof Releasable) {
                 ((Releasable) content).close();
             }
+            transport.inFlightRequestsBreaker.addWithoutBreaking(-reservedBytes);
```

Comment:
I now throw an `IllegalStateException` in case we attempt to send a response on a closed channel or if we try to close a channel twice.